### PR TITLE
feat(monorepo): added pool package

### PR DESCRIPTION
**Pull request by prodesert22**
## Summary

- Added pool package

## Tasks
https://app.clickup.com/t/85ztqx0vj

### CODE REVIEW

File: monorepo/packages/wallet-connectors/src/NearConnector/near.ts [hunk 1]
```diff
@@ -61,58 +69,6 @@ function getNearConfig(env = 'testnet') {
   }
 }
 
-export const near = new NearConnector({
-  normalizeChainId: false,
-  normalizeAccount: false,
-  config: getNearConfig('testnet'),
-});
```
Comment (SarjuHansaliya) [inline monorepo/packages/wallet-connectors/src/NearConnector/near.ts:68]: what is the reason to move this to `src/NearConnector/index.ts` ?

File: monorepo/packages/wallet-connectors/src/NearConnector/index.ts [hunk 2]
```diff
@@ -131,3 +138,44 @@ export class NearConnector extends AbstractConnector {
     return false;
   }
 }
+
+function getNearMainnetConfig() {
```
Comment (SarjuHansaliya) [inline monorepo/packages/wallet-connectors/src/NearConnector/index.ts:142]: `getNearMainnetConfig`, `getNearConfig` is already defined in `near.ts`

File: monorepo/packages/wallet-connectors/src/index.tsx [hunk 3]
```diff
@@ -74,6 +81,8 @@ export { HashConnector, HashConnectEvents, hashconnectEvent, WalletConnectConnec
 export { UserRejectedRequestError, NoEthereumProviderError };
 export type { WalletConnectConnectorArguments };
 
+export { near };
 export * from './NearConnector/near';
```
Comment (SarjuHansaliya) [inline monorepo/packages/wallet-connectors/src/index.tsx:85]: Actually this is not needed, I am not sure why you moved `near` from `near.ts` to `index.ts`. If its justifiable then lets keep it here or else lets revert all these changes.

File: monorepo/packages/pools/src/hooks/sorting.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import { Token } from '@pangolindex/sdk';
+import { getTokenComparator } from '@pangolindex/shared';
+import { useAllTokenBalances } from '@pangolindex/state-hooks';
+import { useMemo } from 'react';
+
+export function useTokenComparator(inverted: boolean): (tokenA: Token, tokenB: Token) => number {
```
Comment (SarjuHansaliya) [inline monorepo/packages/pools/src/hooks/sorting.ts:6]: We are already using `useTokenComparator` in swap also so we need to move this to somewhere in `state-hooks` something like in `state-hooks/src/state/wallet` so we dont need to manage 2 copies of same hook.

File: monorepo/packages/pools/src/hooks/mint/atom.ts
Comment (SarjuHansaliya) [inline monorepo/packages/pools/src/hooks/mint/atom.ts:1]: lets move `burn` and `mint` to `state` folder to keep it consistent

File: monorepo/packages/pools/package.json [hunk 5]
```diff
@@ -0,0 +1,78 @@
+{
+  "name": "@pangolindex/pool",
```
Comment (SarjuHansaliya) [inline monorepo/packages/pools/package.json:2]:
```suggestion
  "name": "@pangolindex/pools",
```

**Review by SarjuHansaliya — CHANGES_REQUESTED** [review]
- [ ] make sure `yarn lint` works properly and commit formatted files
- [ ] make sure `yarn build:packages` works

File: monorepo/packages/wallet-connectors/src/index.tsx [hunk 6]
```diff
@@ -66,6 +67,12 @@ export const venly = new VenlyConnector({
   supportedChainIds: SUPPORTED_EVM_CHAINS_ID,
 });
 
+export const hashConnect = new HashConnector({
```
[the remark this answers is not in the document]
Comment (prodesert22) [reply monorepo/packages/wallet-connectors/src/index.tsx]: This changes are made before you fix the wallet-connectors package, I'm reverting now

File: monorepo/packages/wallet-connectors/src/NearConnector/index.ts [hunk 7]
```diff
@@ -1,8 +1,15 @@
 import { JsonRpcProvider } from '@ethersproject/providers';
+import { ChainId } from '@pangolindex/sdk';
 import { AbstractConnector } from '@web3-react/abstract-connector';
 import { AbstractConnectorArguments } from '@web3-react/types';
 import { Near, WalletConnection, keyStores } from 'near-api-js';
 
+// Near Exchnage Contract
+export const NEAR_EXCHANGE_CONTRACT_ADDRESS = {
+  [ChainId.NEAR_MAINNET]: 'png-exchange-v1.mainnet',
```
[the remark this answers is not in the document]
Comment (prodesert22) [reply monorepo/packages/wallet-connectors/src/NearConnector/index.ts]: This changes are made before you fix the wallet-connectors package, I'm reverting now

File: monorepo/packages/wallet-connectors/src/NearConnector/near.ts (hunk 1 above)
Comment (prodesert22) [reply monorepo/packages/wallet-connectors/src/NearConnector/near.ts]: same of above

File: monorepo/packages/wallet-connectors/src/NearConnector/index.ts (hunk 7 above)
[the remark this answers is not in the document]
Comment (prodesert22) [reply monorepo/packages/wallet-connectors/src/NearConnector/index.ts]: same of above

File: monorepo/packages/wallet-connectors/src/NearConnector/index.ts (hunk 2 above)
Comment (prodesert22) [reply monorepo/packages/wallet-connectors/src/NearConnector/index.ts]: same of above

**Review by SarjuHansaliya — APPROVED** [review]

**Pull request merged**